clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -395,7 +395,7 @@ else if (metadata.error() == Errors.TOPIC_AUTHORIZATION_FAILED)
             Integer currentEpoch = lastSeenLeaderEpochs.get(tp);
             if (topicId != null && oldTopicId != null && !topicId.equals(oldTopicId)) {
                 // If both topic IDs were valid and the topic ID changed, update the metadata
-                log.debug("Topic ID for partition {} changed from {} to {}, so this topic must have been recreated. " +
+                log.info("Topic ID for partition {} changed from {} to {}, so this topic must have been recreated. " +
```

Comment:
This is very fussy, but for some reason, the phrasing here is bugging me. The addition of "must have" almost makes the event seem more uncertain and open to interpretation. Like we need to reassure the user that our deduction is correct. Maybe we can leave that part out?
```java
log.info("Resetting the last seen epoch of partition {} to {} since the associated topicId changed from {} to {}"...
```